cmd/descheduler/app/version.go

```diff
@@ -81,7 +77,19 @@ func splitVersion(version string) (string, string) {
 	if version == "" {
 		return "", ""
 	}
-	// A sample version would be of form v0.1.0-7-ge884046, so split at first '.' and
-	// then return 0 and 1+(+ appended to follow semver convention) for major and minor versions.
-	return strings.Trim(strings.Split(version, ".")[0], "v"), strings.Split(version, ".")[1] + "+"
+
+	// Version from an automated container build environment for a tag. For example v20200521-v0.18.0.
+	if matched, _ := regexp.MatchString(`^v\d{8}-v\d+\.\d+\.\d+$`, version); matched {
+		semVer := strings.Split(version, "-")[1]
+		return strings.Trim(strings.Split(semVer, ".")[0], "v"), strings.Split(semVer, ".")[1] + "+"
+	}
+
+	// Version from an automated container build environment(not a tag) or a local build. For example v20201009-v0.18.0-46-g939c1c0.
+	if matched, _ := regexp.MatchString(`^v\d{8}-v\d+\.\d+\.\d+-\w+-\w+$`, version); matched {
+		semVer := strings.Split(version, "-")[1]
+		return strings.Trim(strings.Split(semVer, ".")[0], "v"), strings.Split(semVer, ".")[1] + "+"
```

Comment:
Any plans to parse the string differently? Right now both if blocks are identical.

Reply:
Good catch. I'll take a look at this in a few days and refactor it to remove the duplicated code.

Reply:
I refactored in commit 4cc5dc9004b701172496ce4416967c939470c401. @ingvagabund let me know if you think this looks better or not. Thanks!